cfme/automate/dialog_box.py

```diff
@@ -0,0 +1,111 @@
+
+from utils.appliance import Navigatable
+from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
+
+
+from .dialog_tab import AddTabView
+
+
+class BoxForm(AddTabView):
+    box_label = Input(name='group_label')
+    box_desc = Input(name="group_description")
+
+
+class AddBoxView(BoxForm):
+    """AddBox View.
+       Args:
+             BoxForm.
+    """
+    plus_btn = Dropdown('Add')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_customization and self.service_dialogs.is_opened and
+            self.title.text == "Adding a new Dialog [Box Information]"
+        )
+
+
+class EditBoxView(BoxForm):
+    """EditBox View.
+       Args:
+             BoxForm.
+    """
+    @property
+    def is_displayed(self):
+        return (
+            self.in_customization and self.service_dialogs.is_opened and
+            self.title.text == "Editing Dialog {} [Box Information]".format(self.box_label)
+        )
+
+
+class BoxCollection(Navigatable):
+    def __init__(self, parent):
+        self.parent = parent
+        Navigatable.__init__(self, appliance=parent.appliance)
+
+    @property
+    def tree_path(self):
+        return self.parent.tree_path
+
+    def instantiate(self, box_label=None, box_desc=None):
+        return Box(self,
+            box_label=box_label, box_desc=box_desc)
+
+    def create(self, box_label=None, box_desc=None):
+        """Create box method.
+           Args:
+             box_label and box_description.
+        """
+        view = navigate_to(self, "AddBox")
+        view.plus_btn.item_select("Add a new Box to this Tab")
+        box_view = navigate_to(self, "Add")
+        fill_dict = {
+            k: v
+            for k, v in {'box_label': box_label, 'box_desc': box_desc}.items()
+            if v is not None}
+        box_view.fill(fill_dict)
+        return self.instantiate(box_label=box_label, box_desc=box_desc)
+
+
+class Box(Navigatable):
+    """A class representing one Box of dialog."""
+    def __init__(self, collection, box_label, box_desc):
+        Navigatable.__init__(self, appliance=collection.appliance)
+        self.collection = collection
+        self.box_label = box_label
+        self.box_desc = box_desc
+
+    @property
+    def parent(self):
+        """returns the parent object - Tab"""
+        return self.collection.parent
+
+    @property
+    def box(self):
+        return BoxCollection(self)
+
+    @cached_property
+    def dialog_element(self):
+        from .dialog_element import ElementCollection
+        return ElementCollection(self)
+
+    @property
+    def tree_path(self):
+        return self.collection.tree_path + [self.box_label]
+
+    @property
+    def tab(self):
+        return self.parent.tab
+
+
+@navigator.register(BoxCollection)
+class AddBox(CFMENavigateStep):
+    VIEW = AddTabView
+
+
+@navigator.register(BoxCollection)
+class Add(CFMENavigateStep):
+    VIEW = AddBoxView
```

Comment:
No navigation required as it automatically navigates to this View after clicking "add box" button